subprojects/wrapper/src/main/java/org/gradle/wrapper/Download.java

```diff
@@ -55,12 +56,73 @@ public Download(Logger logger, DownloadProgressListener progressListener, String
 
-    private void configureProxyAuthentication() {
-        if (System.getProperty("http.proxyUser") != null) {
-            Authenticator.setDefault(new ProxyAuthenticator());
+    private void populateProxyPropertiesFromEnvironmentVariable(String protocol) {
+        // Do not fallback to environment variables if a proxy has been set via properties.
+        if (System.getProperty(protocol + ".proxyHost") != null) {
+            return;
+        }
+
+        String variableName = protocol + "_proxy";
+        String proxyString = System.getenv(variableName);
+        if (proxyString == null) {
+            proxyString = System.getenv(variableName.toUpperCase());
+        }
+
+        try {
+            URL proxyUrl = new URL(proxyString);
+            System.setProperty(protocol + ".proxyHost", proxyUrl.getHost());
+
+            int port = proxyUrl.getPort();
+            if (port > 0) {
+                System.setProperty(protocol + ".proxyPort", String.valueOf(port));
+            }
+
+            String userInfo = proxyUrl.getUserInfo();
+            if (userInfo != null) {
+                String[] userAndPassword = userInfo.split(":", 2);
+                System.setProperty(protocol + ".proxyUser", userAndPassword[0]);
+                if (userAndPassword.length > 1) {
+                    System.setProperty(protocol + ".proxyPassword", userAndPassword[1]);
+                }
+            }
+        } catch (MalformedURLException e) {
+            // Simply ignore a malformed proxy URL.
+            if (proxyString != null) {
+                logger.log("Unable to parse value '" + proxyString + "' of environment variable '" + variableName + "' as a URL.");
+                if (!proxyString.startsWith("http")) {
+                    logger.log("Please ensure it contains a valid protocol prefix.");
+                }
+            }
         }
     }
 
+    private void configureProxyAuthentication() {
+        populateProxyPropertiesFromEnvironmentVariable("http");
+        populateProxyPropertiesFromEnvironmentVariable("https");
+        Authenticator.setDefault(getProxyPasswordAuthenticator());
+    }
+
+    private Authenticator getProxyPasswordAuthenticator() {
+        return new Authenticator() {
+            @Override
+            protected PasswordAuthentication getPasswordAuthentication() {
+                // While Java itself only defines the use of the "http(s).proxyHost" and "http(s).proxyPort" properties, see [1] for Java 8, the Apache HttpClient established the use of the
+                // "http(s).proxyUser" and "http(s).proxyPassword" properties, see [2], so support them here.
+                // [1] https://docs.oracle.com/javase/8/docs/api/java/net/doc-files/net-properties.html#Proxies
+                // [2] https://hc.apache.org/httpcomponents-client-5.0.x/httpclient5/xref/org/apache/hc/client5/http/impl/auth/SystemDefaultCredentialsProvider.html#L117
+
+                String protocol = getRequestingScheme();
```

Comment:
❌ `getRequestingScheme()` returns the authentication type (e.g. `basic`) instead of the protocol. Should be: 

```suggestion
                String protocol = getRequestingProtocol();
```

Reply:
Thanks for catching this, fixed.